zerver/lib/bugdown/__init__.py

```diff
@@ -976,9 +994,16 @@ def run(self, root: Element) -> None:
                       class_attr=class_attr,
                       already_thumbnailed=True)
                 continue
+
             if self.is_image(url):
+                image_source = self.corrected_image_source(url)
+                if image_source is not None:
+                    corrected_url = ResultWithFamily(found_url.family, (image_source, image_source))
+                    self.handle_image_inlining(root, corrected_url)
+                    continue
                 self.handle_image_inlining(root, found_url)
                 continue
```

Comment:
I think this would be cleaner using `found_url = ResultWithFamily(found_url.family, (image_source, image_source))` and deduplicating the `handle_image_inlining` calls.